deploy-service/common/src/main/java/com/pinterest/deployservice/handler/PingHandler.java

```diff
@@ -640,7 +640,7 @@ public PingResult ping(PingRequestBean pingRequest, boolean rate_limited) throws
 
         // Now we have all the relevant envs, reports and agents, let us do some
         // analysis & pick the potential install candidate and uninstall candidate
-        GoalAnalyst analyst = new GoalAnalyst(deployDAO, environDAO, hostName, hostId, envs, reports, agents);
+        GoalAnalyst analyst = new GoalAnalyst(deployConstraintDAO, hostTagDAO, deployDAO, environDAO, hostName, hostId, envs, reports, agents, ec2Tags);
```

Comment:
instead of having the update in GoalAnalyst, can we have a separate method for updateHostTags() similar to updateHostStatus and updateHosts

Reply:
nope, we need to update per envId. And the envId logic is in GoalAnalyst instead of PingHandler